tools/shared/src/main/scala/org/scalajs/core/tools/linker/analyzer/Analyzer.scala

```diff
@@ -416,47 +416,43 @@ private final class Analyzer(semantics: Semantics,
       if (!allowAddingSyntheticMethods) {
         tryLookupMethod(proxyName)
       } else {
-        /* The lookup for a target method in this code implements the
-         * algorithm defining `java.lang.Class.getMethod`. This mimics how
-         * reflective calls are implemented on the JVM, at link time.
-         *
-         * Caveat: protected methods are not ignored. This can only make an
-         * otherwise invalid reflective call suddenly able to call a protected
-         * method. It never breaks valid reflective calls. This could be fixed
-         * if the IR retained the information that a method is protected.
-         */
-
-        @tailrec
-        def loop(ancestorInfo: ClassInfo): Option[MethodInfo] = {
-          if (ancestorInfo ne null) {
-            ancestorInfo.methodInfos.get(proxyName) match {
-              case Some(m) =>
-                assert(m.isReflProxy && !m.isAbstract)
-                Some(m)
-
-              case _ =>
-                ancestorInfo.findProxyMatch(proxyName) match {
-                  case Some(target) =>
-                    val targetName = target.encodedName
-                    Some(ancestorInfo.createReflProxy(proxyName, targetName))
-
-                  case None =>
-                    loop(ancestorInfo.superClass)
-                }
-            }
-          } else {
-            None
+        methodInfos.get(proxyName).orElse {
+          findReflectiveTarget(proxyName).map { reflectiveTarget =>
+            createReflProxy(proxyName, reflectiveTarget.encodedName)
```

Comment:
Do we really want to create the reflexive proxy on the leaf node in the hierarchy? IIUC, if we have a base class `Base` with a method `foo` and three subclasses `A`, `B`, `C, that do not override `foo`, a call to `foo` on `Base` would generate a reflexive proxy in every class that is instantiated.